airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTestsKube.java

```diff
@@ -0,0 +1,332 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.test.acceptance;
+
+import static io.airbyte.api.client.model.ConnectionSchedule.TimeUnitEnum.MINUTES;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import io.airbyte.api.client.AirbyteApiClient;
+import io.airbyte.api.client.invoker.ApiClient;
+import io.airbyte.api.client.invoker.ApiException;
+import io.airbyte.api.client.model.CheckConnectionRead;
+import io.airbyte.api.client.model.ConnectionCreate;
+import io.airbyte.api.client.model.ConnectionIdRequestBody;
+import io.airbyte.api.client.model.ConnectionRead;
+import io.airbyte.api.client.model.ConnectionSchedule;
+import io.airbyte.api.client.model.ConnectionStatus;
+import io.airbyte.api.client.model.ConnectionUpdate;
+import io.airbyte.api.client.model.DestinationCreate;
+import io.airbyte.api.client.model.DestinationIdRequestBody;
+import io.airbyte.api.client.model.DestinationRead;
+import io.airbyte.api.client.model.JobInfoRead;
+import io.airbyte.api.client.model.JobStatus;
+import io.airbyte.api.client.model.SourceCreate;
+import io.airbyte.api.client.model.SourceIdRequestBody;
+import io.airbyte.api.client.model.SourceRead;
+import io.airbyte.api.client.model.SourceSchema;
+import io.airbyte.api.client.model.SyncMode;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.persistence.PersistenceConstants;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import org.junit.Before;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.testcontainers.shaded.org.apache.commons.lang.RandomStringUtils;
+
+/**
+ * These tests are a limited form of the tests in {@link AcceptanceTests}. They are only testing for
+ * a lack of API failures. For now, we aren't re-using {@link AcceptanceTests} because we would need
```

Comment:
Is the issue that AcceptanceTests uses a docker container? If so, would it make sense to use ERIO there?

Reply:
It uses a Docker container that exists outside of Kube, so we would have to either provide a tunnel for the Kube service to access the localhost (non-Kube) DB or we would need to pakcage the docker operations into Kube. Longer term we should do both. Short term this seemed sufficient.

Reply:
Should we just stop using a container db and use ERIO instead?

Reply:
I think I would prefer Sherif's suggestion here, if the only change is that we are switching out the source, then I would prefer we use the same AcceptanceTest class for both kube and docker versions.  I'm fine with us just using exchange rate api for both of them. AcceptanceTest is a big class to just copy and paste. We also lose the value of having acceptance tests a little bit if kube and docker have different acceptance tests (or if those acceptance tests drift apart meaningfully, which i assume will happen.) 

If I'm missing something more complex in here, lmk.

Reply:
I think it's much more valuable to have Postgres as a source for incremental testing, more data types, incremental vs full refresh, the potential for customization of data type transformations, etc. 

The new class is 1/2 the size of the other. 

Really I think of this is a stopgap until we add support for different types of container interactions (docker and kube) in the main acceptance test class. Then we could run it in both.